Documentation updates for 1.50.0 (#3002)
FusionAuth/fusionauth-site: astro/src/content/docs/lifecycle/authenticate-users/oauth/endpoints.mdx

If you will be using the Refresh Token Grant, you will make a request to the Token endpoint to exchange the user's refresh token for an access token.

> <API method="POST" uri="/oauth2/token"/>

Reviewer: Was this all duplicate?

Author: Yes. The entire `Refresh Token Grant Request` request portion was duplicated.

  </APIField>
</APIBlock>

> #### Request Headers

Reviewer: Was this section duplicated?

Author: Yes. It duplicated the request headers and parameters from the `Validate and retrieve a user_code using client credentials` example after providing the correct headers and parameters for using an API key.

{

>   "client_id": "3c219e58-ed0e-4b18-ad48-f4f92793ae32",

Reviewer: Should probably break these out to json files and include them. Not sure if appropriate to do now.

}
```

## UserInfo

> This is an implementation of the OpenId Connect UserInfo endpoint as defined by the [OpenId Connect Section 5.3](http://openid.net/specs/openid-connect-core-1_0.html#UserInfo).

Reviewer: Can you update DocsDevReadme to specify it should be `UserInfo` not `Userinfo`?

    ```
    Authorization: Bearer [access_token]
    ```

    <Aside type="caution">

>       In version `1.50.0` and later, the `aud` claim is required for all tokens. The `aud` claim value, or the first value in the case the claim contains a list, must correspond to an active application.

Reviewer: ```suggestion
      In version `1.50.0` and later, the `aud` claim is required for all tokens. The `aud` claim value, or the first value in the case where the claim contains a list, must correspond to an active Application.
```

      If you have a token that was not obtained from an OAuth flow and does not contain the `aud` claim, consider adding the claim or using the [JWT Validate API](/docs/apis/jwt#validate-a-jwt) instead.
    </Aside>

    <Aside type="note">

>       In version `1.50.0` and later when the application's <InlineField>oauthConfiguration.scopeHandlingPolicy</InlineField> is set to `Strict`, the provided token must:

Reviewer: ```suggestion
      In version `1.50.0` and later, when the application's <InlineField>oauthConfiguration.scopeHandlingPolicy</InlineField> is set to `Strict`, to use a token with this endpoint, the token must:
```

>       * Be an access token rather than an id token

Reviewer: ```suggestion
      * Be an access token
```

      * Contain the `openid` scope in the `scope` claim

>       in order to be used with this endpoint.

Reviewer: ```suggestion
```

    </Aside>
  </APIField>
</APIBlock>

### Response

<Aside type="note">
  In version `1.50.0` and later the UserInfo response changes based on the <InlineField>oauthConfiguration.scopeHandlingPolicy</InlineField> value of the application object.

  * In `Strict` mode the response claims are populated based on the `scope` values of the provided token and available information on the User object. This behavior more closely aligns with specification recommendations.

>   * In `Compatibility` mode the response claims are populated based on available information on the User object only and include some claims from the provided token. This is legacy behavior maintained for backwards compatibility.

Reviewer: This list gets repeated a lot, maybe worth extracting to a component?

>   * In `Compatibility` mode the response claims are populated based on available information on the User object only and include some claims from the provided token. This is legacy behavior maintained for backwards compatibility.

Author: What list are you referring to?

>   * In `Compatibility` mode the response claims are populated based on available information on the User object only and include some claims from the provided token. This is legacy behavior maintained for backwards compatibility.

Reviewer: The list of options for the "Scope handling policy"

>   * In `Compatibility` mode the response claims are populated based on available information on the User object only and include some claims from the provided token. This is legacy behavior maintained for backwards compatibility.

Author: I was confused because you called it out in the one place where the wording is slightly different than the others.

How far down the component rabbit hole do we want to go? I don't see other cases where enum options are split into their own component. I think that Astro is much easier to parse than ASCIIdoc, but we had some issues with includes making it hard to follow.

If we add a component for these options, what about `Relationship`, `Consent mode`, `Unknown scope policy`, etc.?

>   * In `Compatibility` mode the response claims are populated based on available information on the User object only and include some claims from the provided token. This is legacy behavior maintained for backwards compatibility.

Reviewer: Yeah, it's a balance. I just know when I was making suggestions, I had to change the wording in like 8 places. 

I know I created components for some lists ([JWT algorithm](https://github.com/FusionAuth/fusionauth-site/blob/master/astro/src/content/docs/apis/_jwt_algorithm.mdx)) but we haven't been consistent.

Let's leave it be for now.

    * `change_password_validation`
    * `client_authentication_missing`
    * `client_id_mismatch`
    * `consent_canceled`
    * `grant_type_disabled`
    * `invalid_additional_client_id`
    * `invalid_client_authentication_scheme`
    * `invalid_client_authentication`
    * `invalid_client_authentication_scheme`
    * `invalid_client_id`
    * `invalid_device_code`
    * `invalid_entity_permission_scope`
    * `invalid_grant_type`
    * `invalid_id_token_hint`
    * `invalid_origin`

>     * `invalid_pkce_code_challenge_method`

Reviewer: Is that a breaking change?

Author: No. I just moved it down two rows to be in alphabetical order.

    * `invalid_origin_opaque`
    * `invalid_pkce_code_challenge`
    * `invalid_pkce_code_challenge_method`
    * `invalid_pkce_code_verifier`
    * `invalid_post_logout_redirect_uri`
    * `invalid_redirect_uri`
    * `invalid_response_mode`
    * `invalid_response_type`
    * `invalid_target_entity_scope`
    * `invalid_user_code`

>     * `invalid_user_id`

Reviewer: is that a breaking change?

Author: No. I just moved it down two rows to be in alphabetical order.
